src/volume.jl

```diff
@@ -0,0 +1,310 @@
+            0 0 1 0
+            0 0 0 1
+        ]),
+        @SMatrix([  # translate
+            1 0 0 (r + l)/2n
+            0 1 0 (t + b)/2n
+            0 0 1 0
+            0 0 0 1
+        ]),
+        @SMatrix([  # perspective
+            -1 0 0 0  # flip x
+            0 -1 0 0  # flip y
+            0 0 (f + n)/(f - n) -2f * n/(f - n)
+            0 0 1 0
+        ]),
+    )
+end
+
+"""
+    ortho(l, r, b, t, n, f)
+
+# Description
+
+Computes the orthographic projection matrix.
+
+# Arguments
+
+    - `l`: left coordinate of the vertical clipping plane.
+    - `r` : right coordinate of the vertical clipping plane.
+    - `b`: bottom coordinate of the horizontal clipping plane.
+    - `t`: top coordinate of the horizontal clipping plane.
+    - `n`: distance to the near depth clipping plane.
+    - `f`: distance to the far depth clipping plane.
+"""
+ortho(l, r, b, t, n, f) = *(
+    @SMatrix([  # scale
+        2/(r - l) 0 0 0
+        0 2/(t - b) 0 0
+        0 0 2/(f - n) 0
+        0 0 0 1
+    ]),
+    @SMatrix([  # translate
+        1 0 0 -(l + r)/2
+        0 1 0 -(t + b)/2
+        0 0 1 -(f + n)/2
+        0 0 0 1
+    ]),
+)
+
+abstract type Projection end
+
+struct Orthographic{T} <: Projection where {T}
+    A::SMatrix{4,4,T}
+    Orthographic(args::T...) where {T} = new{float(T)}(ortho(args...))
+end
+
+struct Perspective{T} <: Projection where {T}
+    A::SMatrix{4,4,T}
+    Perspective(args::T...) where {T} = new{float(T)}(frustum(args...))
+end
+
+function ctr_len_diag(x, y, z)
+    mx, Mx = NaNMath.extrema(float.(x))
+    my, My = NaNMath.extrema(float.(y))
+    mz, Mz = NaNMath.extrema(float.(z))
+
+    lx = Mx - mx
+    ly = My - my
+    lz = Mz - mz
+
+    (
+        @SVector([mx + 0.5lx, my + 0.5ly, mz + 0.5lz]),
+        @SVector([mx, my, mz]),
+        @SVector([Mx, My, Mz]),
+        @SVector([lx, ly, lz]),
+        √(lx^2 + ly^2 + lz^2),
+    )
+end
+
+cube_corners(mx, Mx, my, My, mz, Mz) = [
+    mx mx mx mx Mx Mx Mx Mx
+    my my My My my my My My
+    mx Mz mx Mz mx Mz mx Mz
+]
+
+function view_matrix(center, distance, elevation, azimuth, up)
+    up_str = string(up)
+    shift = if (up_axis = Symbol(up_str[end])) === :x
+        0
+    elseif up_axis === :y
+        1
+    elseif up_axis === :z
+        2
+    else
+        throw(error("$up not understood"))
+    end
+    up_vector = circshift(
+        [
+            length(up_str) == 1 ? 1 : (p = +1, m = -1)[Symbol(up_str[1])]
+            0
+            0
+        ],
+        shift,
+    )
+    cam_move = circshift(
+        distance .* [
+            sind(elevation)
+            cosd(azimuth) * cosd(elevation)
+            sind(azimuth) * cosd(elevation)
+        ],
+        shift,
+    )
+    lookat(center .+ cam_move, center, up_vector)
+end
+
+"""
+    MVP(x, y, z; $(keywords(; default = (), add = (:x, :y, :z, :projection, :elevation, :azimuth, :zoom, :up))))
+
+# Description
+
+Model - View - Projection transformation matrix.
+"""
```

Comment:
Since this is an exported symbol, it would be helpful to have a few examples and "see-also" links here so that people can quickly understand what this symbol is for. It would also be beneficial to have some links to external resources, e.g., http://www.codinglabs.net/article_world_view_projection_matrix.aspx

Reply:
I've added some theory links. As for the examples, there are some "plot by hand" examples (using `lineplot` and `MVP`) in `test/tst_volume.jl`.